### DISCUSSION TITLE
UPS read the request and response string

### DISCUSSION THREAD

**Discussion by ebaj-tech** [alexeybusygin/ShippingRates#77] [Q&A] [answered]
Is there a way to read the request and response string? This makes it easier to communicate with UPS customer service. Customer service often wants to know what we are requesting or what we are receiving as a response.

**Comment by alexeybusygin** [comment]
Right now, requests and responses cannot be read. However, injecting ILogger for this purpose sounds like a good idea, so I will create an issue for that.

**Comment by alexeybusygin** [comment, marked as answer]
Implemented in v2.3.1.
